src/CrudTrait.php

```diff
@@ -191,4 +191,215 @@ public function uploadMultipleFilesToDisk($value, $attribute_name, $disk, $desti
+                    case 'image/png':
+                        $extension = '.png';
+                        break;
+                    default:
+                        $extension = '.jpg';
+                        break;
+                }
+
+                $variant_name = $new_file_name.'-'.$variant.$extension;
+                $variant_file = $destination_path.'/'.$variant_name;
+
+                if ($dimensions) {
+                    $width = $dimensions[0];
+                    $height = $dimensions[1];
+
+                    if ($img->width() > $width || $img->height() > $height) {
+                        $img->resize($width, $height, function ($constraint) {
+                            $constraint->aspectRatio();
+                        })
+                        ->save($disk_root.'/'.$variant_file);
+                    } else {
+                        $img->save($disk_root.'/'.$variant_file);
+                    }
+
+                    $image_variations[$variant] = $public_path.'/'.$variant_file;
+                } else {
+                    $img->save($disk_root.'/'.$variant_file);
+                    $image_variations['original'] = $variant_file;
+                }
+            }
+
+            return $this->attributes[$attribute_name] = $image_variations['original'];
+        }
+    }
+
+    /**
+     * Handles the retrieval of an image by variant:.
+     *
+     * @param  [type] $attribute        Name of the attribute within the model that contains the json
+     * @param  [type] $variant          Name of the variant you want to extract
+     * @param  [type] $disk             Filesystem disk used to store files.
+     */
+    public function getUploadedImageFromDisk($attribute, $variant = 'original', $disk = null)
+    {
+        $image = $this->attributes[$attribute];
+        $url = null;
+        if (! empty($image)) {
+            $image_variant = str_replace('-original', '-'.$variant, $image);
+
+            if ($disk) {
+                $disk_config = config('filesystems.disks.'.$disk);
+                $disk_root = $disk_config['root'];
+
+                //if the disk is public, we need to know the public path
+                if ($disk_config['visibility'] == 'public') {
+                    $public_path = str_replace(public_path(), '', $disk_root);
+                } else {
+                    $public_path = $disk_root;
+                }
+
+                if (\Storage::disk($disk)->exists($image_variant)) {
+                    $url = asset($public_path.'/'.$image_variant);
+                } else {
+                    $url = asset($public_path.'/'.trim($image, '/'));
+                }
+            } else {
+                if (\Storage::exists($image_variant)) {
+                    $url = \Storage::url(trim($image_variant, '/'));
+                } else {
+                    $url = url($image);
+                }
+            }
+        }
+
+        return $url;
```

Comment:
This can be 'null' but can `Storage::url($foo)` ever also be null? I don't think it can in which case it's ok but if there is a difference between:

1. $image was not set for some reason
2. `Storage::url()` returned something funny

...we probably should allow the end user to detect that somehow. Or throw an exception.